src/document/transformoperation.js

```diff
@@ -0,0 +1,434 @@
+/**
+ * @license Copyright (c) 2003-2015, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+'use strict';
+
+/**
+ * Transforms given {document.operation.Operation} by another {document.operation.Operation} and returns the result of
+ * that transformation as an Array containing one or more {document.operation.Operation} elements.
+ *
+ * Operations operate on specified positions, passed to them when they are created. Whenever {@link document.Document document}
+ * changes, we have to reflect those modifications by updating, or "transforming", operations (which are not yet applied).
+ * When operation is transformed, its parameters may change basing on the operation by which it is transformed.
+ * If the transform-by operation applied any modifications to the Tree Model that affects positions or nodes
+ * connected with transformed operation, those changes will be reflected in the parameters of the returned operation(s).
+ *
+ * Whenever the {@link document.Document document} has different {@link document.Document#baseVersion}
+ * than an operation you want to {@link document.Document#applyOperation apply}, you need to transform that
+ * operation by all the operations that were applied to the {@link document.Document document} since it has
+ * {@link document.Document#baseVersion} same as the operation. Transform them in the same order as those
+ * operations were applied. This way all modifications done to the Tree Model will be reflected
+ * in the operation parameters and the operation will "operate" on "up-to-date" version of the Tree Model.
+ * This is mostly the case with Operational Transformation but it might be needed in particular features.
+ *
+ * In some cases, when given operation apply changes to the same nodes as this operation, there is a need
+ * to create two operations as a result. It would be impossible to create just one operation that handles
+ * modifications needed to be applied to the tree. This is why Array is returned instead of single object.
+ * All returned operations has to be applied (or further transformed) to get an effect that was intended in
+ * pre-transformed operation.
+ *
+ * @function transformOperation
+ * @param {document.operation.Operation} a Operation that will be transformed.
+ * @param {document.operation.Operation} b Operation to transform by.
+ * @param {Boolean} isStrong Flag indicating whether this operation should be treated as more important
+ * when resolving conflicts.
```

Comment:
I think you could add one more paragraph about the `isStrong` parameter. Explain why we need this parameter. Explain why we can not assume that operation a is always stronger and switch the order of the parameters when we need. In fact it makes no sense, but I believe that this is the question one will ask when he will read this code for the first time.